spec/ComputerLogicSpec.js

```diff
@@ -1,41 +1,118 @@
+var WinConditions = require ('../src/WinConditions');
 var ComputerLogic = require ('../src/ComputerLogic');
 
 describe("ComputerLogic", function() {
 
   beforeEach(function() {
-    computerLogic = new ComputerLogic();
+    winConditions = new WinConditions();
+    computerLogic = new ComputerLogic(winConditions);
   });
 
-  describe("#getMove", function() {
-    it("will return the board", function() {
+  describe("#pickSpace", function() {
+    it ("will return the center position if it is not chosen by the player", function() {
+      var gameBoard = ["X","1","2","3","4","5","6","7","8"];
+      expect(computerLogic.pickSpace(gameBoard)).toEqual(4);
+    });
+    it ("will block a top horizontal win", function() {
+      var gameBoard = ["X","X","2","O","4","5","6","7","8"];
+      expect(computerLogic.pickSpace(gameBoard)).toEqual(2);
+    });
+    it ("will block a middle horizontal win", function() {
+      var gameBoard = ["0","1","O","3","X","X","6","7","8"];
+      expect(computerLogic.pickSpace(gameBoard)).toEqual(3);
+    });
+    it ("will block a bottom horizontal win", function() {
+      var gameBoard = ["0","1","2","3","O","5","6","X","X"];
+      expect(computerLogic.pickSpace(gameBoard)).toEqual(6);
+    });
+    it ("will block a left vertical win", function() {
+      var gameBoard = ["X","1","2","X","O","5","6","7","8"];
+      expect(computerLogic.pickSpace(gameBoard)).toEqual(6);
+    });
+    it ("will block a middle vertical win", function() {
+      var gameBoard = ["0","X","2","3","X","5","6","7","O"];
+      expect(computerLogic.pickSpace(gameBoard)).toEqual(7);
+    });
+    it ("will block a right vertical win", function() {
+      var gameBoard = ["0","1","X","3","O","5","6","7","X"];
+      expect(computerLogic.pickSpace(gameBoard)).toEqual(5);
+    });
+    it ("will block a backward diagonal win", function() {
+      var gameBoard = ["0","1","2","O","X","5","6","7","X"];
+      expect(computerLogic.pickSpace(gameBoard)).toEqual(0);
+    });
+    it ("will block a forward diagonal win", function() {
+      var gameBoard = ["0","1","X","3","X","5","6","7","O"];
+      expect(computerLogic.pickSpace(gameBoard)).toEqual(6);
+    });
+    it ("will place the token to win if there is no imminent loss", function() {
+      var gameBoard = ["O","X","X","3","4","5","X","7","O"];
+      expect(computerLogic.pickSpace(gameBoard)).toEqual(4);
+    });
+    it ("will play a game and O will win or tie", function() {
+
+      play = function(gameBoard) {
+        while (! winConditions.isGameOver(gameBoard) ) {
+          var gameBoardSpace = Math.floor(Math.random() * 10);
+          gameBoard[gameBoardSpace] = "X";
+          var computerMove = computerLogic.pickSpace(gameBoard);
+          gameBoard[computerMove] = "O";
+        }
+        return gameBoard;
+      }
+
+      computerWinOrTie = function(endOfGameBoard) {
```

Comment:
Is there a reason you don't use WinCondition here?

Reply:
WinCondition only looks for a Win and it does not care about who wins. So in order to ensure that 'O' is the winner, I needed to make my own method.

Reply:
So I understand that's why it looks like this, but do you think it may be a sign that potentially a way to figure out _WHO_ won would be helpful? Whether it be `checkMarkerForWin('X',gameBoard)` & `checkMarkerForWin('O',gameBoard)` or `getWinningMarker(gameBoard)`, etc.